### DISCUSSION TITLE
Do edge-features need to be the same for direct and reverse edges for a graph to be undirected?

### DISCUSSION THREAD

**Discussion by errhernandez** [pyg-team/pytorch_geometric#8293] [Q&A] [answered]
Hi! I am constructing graphs that judging from the edge_index only should be undirected, because for every edge pointing from node i to node j there is always an edge pointing from j to i. However, it is possible that the edge attributes of edge i->j are different from those of the edge j->i. Does this make the graph not qualify as undirected? In other words, is the criterium for setting a graph as directed/undirected dependent on edge attributes as well as edge indices? Indeed, when I use is_undirected it comes back as false...

I tried to answer this question by checking directly in the source code, but could not interpret how it is done, I'm afraid.

**Comment by rusty1s** [comment, marked as answer]
PyG does not really know about the concept of "undirected graphs", so you are free to either replicate edge features for both directions, or use separate edge features.

Regarding `to_undirected`: Yes, these methods ensure that edge features match in both directions.

**Reply by errhernandez** [reply]
Thank you for your reply. My question was motivated by the fact that looking only edge_index, my graphs should be undirected, but when I use graph.is_undirected() it returns False, so I was wondering if method is_undirected() checks on edge_attr as well as on edge_index to decide if a graph is undirected or not. I guess this is the case.

**Reply by rusty1s** [reply]
Yes, it will also look at edge features if they are present.
